Source/WebCore/platform/graphics/transforms/TransformationMatrix.cpp

```diff
@@ -923,8 +931,8 @@ TransformationMatrix& TransformationMatrix::rotate3d(double x, double y, double
     // Angles are in degrees. Switch to radians.
     angle = deg2rad(angle);
 
-    double sinTheta = sin(angle);
-    double cosTheta = cos(angle);
+    double sinTheta = roundEpsilonToZero(sin(angle));
```

Comment:
Not sure if this should be down here in TransformationMatrix or be the responsibility of callers who care about being close to zero?

Reply:
I think it should be here, or possibly at some point in an inner helper that does degrees -> sin/cos conversion.

This TransformationMatrix function takes an angle in degrees, and wants the rotation matrix for it. The bug is introduced by going via radians before computing the matrix vectors, and losing precision slightly. My take is that's a bug in the implementation detail and no caller would want the other behaviour.